addons/base_import/models/base_import.py

```diff
@@ -1474,7 +1485,7 @@ def execute_import(self, fields, columns, options, dryrun=False):
 
         binary_filenames = self._extract_binary_filenames(import_fields, input_file_data)
 
-        import_fields, merged_data = self._handle_multi_mapping(import_fields, input_file_data)
+        import_fields, merged_data = self.with_context(options=options)._handle_multi_mapping(import_fields, input_file_data)
```

Comment:
It sounds better to have a more verbose context key to avoid conflict.
by example:
```suggestion
        import_fields, merged_data = self.with_context(import_options=options)._handle_multi_mapping(import_fields, input_file_data)
```